airbyte-integrations/connectors/destination-snowflake/src/test/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingSqlOperationsTest.java

```diff
@@ -66,4 +74,24 @@ void removeStage() {
     assertEquals(expectedQuery, actualRemoveQuery);
   }
 
+  @ParameterizedTest
+  @CsvSource({"TEST,false", "but current role has no privileges on it,true"})
+  public void testCreateStageIfNotExists(final String message, final boolean shouldCapture) {
+    final JdbcDatabase db = Mockito.mock(JdbcDatabase.class);
+    final var stageName = "foo";
+    try {
+      Mockito.doThrow(new SnowflakeSQLException(message)).when(db).execute(Mockito.anyString());
+    } catch (SQLException e) {
+      // This would not be expected, but the `execute` method above will flag as an unhandled exception
+      assert false;
+    }
+    final Exception exception = Assertions.assertThrows(Exception.class, () -> snowflakeStagingSqlOperations.createStageIfNotExists(db, stageName));
+    if (shouldCapture) {
+      Assertions.assertInstanceOf(ConfigErrorException.class, exception);
+    } else {
+      Assertions.assertInstanceOf(SnowflakeSQLException.class, exception);
+      Assertions.assertEquals(exception.getMessage(), message);
```

Comment:
Minor nit: you probably don't need to explicitly define `Assertions.assertEquals` since at the top of the imports we import 

```java
import static org.junit.jupiter.api.Assertions.assertEquals;
```

Likely you can also do the same thing with `assertInstanceOf`